### PR TITLE
feat: :sparkles: check merge-request titles for conventional commit

### LINKED ISSUE #42: feat(auto-merge): lint merge request title for conventional commit

### PR DESCRIPTION
**Pull request by anbraten**
closes #42

### CODE REVIEW

File: plugins/auto_merge/checks/is_title_using_conventional_commit.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+	"github.com/GEPROG/lassie-bot-dog/plugins/auto_merge/config"
+	"github.com/xanzy/go-gitlab"
+)
+
+var conventionalCommitSpecUrl = "https://www.conventionalcommits.org/en/v1.0.0/#specification"
+var ConventionalCommitRegex = regexp.MustCompile(`^(feat|fix|docs|style|refactor|perf|test|chore)(?:\((.*)\))?(\!)?\: (.*)$`)
+
+type IsTitleUsingConventionalCommit struct {
+}
```
Comment (lukashass) [inline plugins/auto_merge/checks/is_title_using_conventional_commit.go:14]: Why is this struct empty?

File: plugins/auto_merge/checks/is_title_using_conventional_commit.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+func (check IsTitleUsingConventionalCommit) Check(config *config.AutoMergeConfig, project *gitlab.Project, mergeRequest *gitlab.MergeRequest) bool {
+	return ConventionalCommitRegex.MatchString(mergeRequest.Title)
+}
+
+func (plugin IsTitleUsingConventionalCommit) Name() string {
+	return "is-not-work-in-progress"
```
Comment (lukashass) [inline plugins/auto_merge/checks/is_title_using_conventional_commit.go:21]: I think this has to be changed.

**Review by lukashass — CHANGES_REQUESTED** [review]

File: plugins/auto_merge/checks/is_title_using_conventional_commit.go (hunk 1 above)
Comment (anbraten) [reply plugins/auto_merge/checks/is_title_using_conventional_commit.go:14]: @lukashass The "class" has no internal variables.

File: plugins/auto_merge/checks/is_title_using_conventional_commit.go (hunk 2 above)
Comment (anbraten) [reply plugins/auto_merge/checks/is_title_using_conventional_commit.go]: thx changed it

**Review by lukashass — APPROVED** [review]

**Pull request merged**